### PR TITLE
qemu_guest_agent: Add reboot/halt support - s390x

### PR DESCRIPTION
**Pull request by BohdanMar**
Added Reboot and Halt patterns for s390x. 

ID: 2073295
Signed-off-by: Bogdan Marcynkov bmarcynk@redhat.com

### CODE REVIEW

**Comment by BohdanMar** [comment]
@ngu-niny and @PaulYuuu you please help to review this patch? Thanks

**Comment by BohdanMar** [comment]
After applying the patch 3293, rerun failed cases:
 (1/2) Host_RHEL.m8.u7.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: STARTED
| (1/2) Host_RHEL.m8.u7.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (66.27 s)
 (2/2) Host_RHEL.m8.u7.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: STARTED
/ (2/2) Host_RHEL.m8.u7.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (39.62 s)

**Comment by ngu-niny** [comment]
@BohdanMar Bogdan, besides the summary, better to add a description line in the commit info. Besides, we also need it and the 'ID' in the first comment of this page.

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 1]
```diff
@@ -104,12 +104,16 @@
...
+            s390x:
+                gagent_guest_reboot_pattern = "System Reboot"
         - check_halt:
             gagent_check_type = halt
             # This config would be overriden in guest-os.cfg.
             gagent_guest_shutdown_pattern = "System halted"
             Windows:
                 gagent_guest_shutdown_pattern = ""
+            s390x:
+                gagent_guest_reboot_pattern = "System Halt"
```
Comment (PaulYuuu) [inline qemu/tests/cfg/qemu_guest_agent.cfg:116]: This is the shutdown pattern.

Comment (BohdanMar) [reply qemu/tests/cfg/qemu_guest_agent.cfg:116]: @PaulYuuu, I edited this patch, please have a look

**Comment by ngu-niny** [comment]
@BohdanMar Please add the ID in the patch description(i.e. the first comment) part instead of the commit info.

File: qemu/tests/cfg/qemu_guest_agent.cfg (hunk 1 above)
Comment (ngu-niny) [reply qemu/tests/cfg/qemu_guest_agent.cfg:116]: @BohdanMar Please pay attention to 'gagent_guest_reboot_pattern' and 'gagent_guest_shutdown_pattern'.

**Comment by ngu-niny** [comment]
@BohdanMar The code and description LGTM.

Could you apply the patch and retest the cases then paste the test result in the patch since you have done some code update? Thanks.

File: qemu/tests/cfg/qemu_guest_agent.cfg [hunk 2]
```diff
@@ -104,8 +104,12 @@
             gagent_guest_reboot_pattern = "Restarting system"
             Windows:
                 gagent_guest_reboot_pattern = ""
+            s390x:
+                gagent_guest_reboot_pattern = "System Reboot"
         - check_halt:
             gagent_check_type = halt
+            s390x:
+                gagent_guest_shutdown_pattern = "System Halt"
```
Comment (ngu-niny) [inline qemu/tests/cfg/qemu_guest_agent.cfg:112]: @BohdanMar Bogdan, I missed one thing, the pattern specified in lines 111/112 would be overwritten by line 114, so please move it after that line.

**Comment by BohdanMar** [comment]
The latest result after update:
 (1/2) Host_RHEL.m8.u6.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: STARTED
| (1/2) Host_RHEL.m8.u6.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reboot.s390-virtio: PASS (45.48 s)
 (2/2) Host_RHEL.m8.u6.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: STARTED
/ (2/2) Host_RHEL.m8.u6.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.s390x.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_halt.s390-virtio: PASS (19.46 s)

**Comment by ngu-niny** [comment]
LGTM.

Acked-by: Nini Gu <ngu@redhat.com>

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**